### ISSUE TITLE
How to set awsProfile?

### ISSUE THREAD

**Issue by jkwasniak** [multiplegeorges/vue-cli-plugin-s3-deploy#49]
1) I set awsProfile in vue.config.js
```sh
pluginOptions: {
    s3Deploy: {
      ..
      awsProfile: 'default',
     ..
    },
```
2) In .env.production i set awsProfile to 'production'
```sh
# production profile s3
VUE_APP_S3D_AWS_PROFILE=production
```
3) Documentation says: 
```
Options are set in vue.config.js and overridden on a per-environment basis by .env, .env.staging, .env.production, etc.
```
4) My scripts in package.json look like this
```sh
"scripts": {
    "build-production": "vue-cli-service build --mode production",
    "clean": "rimraf dist",
    "deploy-production": "vue-cli-service s3-deploy --mode production",
  }
```
5) after call 
```sh
yarn deploy-production
```
awsProfile is 'default' and not 'production'. 

Question: What I'm doing wrong?

**Comment by jkwasniak** [comment]
Thanks! Unfortunately, the solution with set AWS_PROFILE does not work.

Strange, direct call works
```sh
"scripts": {
    "deploy": "aws s3 cp dist/ s3://www.filecq.de --recursive --profile production",
  }
```
Note: i use aws-cli/1.16.40 Python/3.7.0 Darwin/18.2.0 botocore/1.12.30

**Comment by multiplegeorges** [comment]
Did you try it with the env var inline like `AWS_PROFILE=production yarn
deploy` ?

**Comment by jkwasniak** [comment]
Yes, It didn't work :-(

**Comment by jkwasniak** [comment]
Can it be that, in the file index. js the corresponding row for overrides of the options.awsProfile is missing?

```js
// Check for environment overrides of the options in vue.config.js.
options.awsProfile = process.env.VUE_APP_S3D_AWS_PROFILE || options.awsProfile
options.region = process.env.VUE_APP_S3D_REGION || options.region
options.bucket = process.env.VUE_APP_S3D_BUCKET || options.bucket
..
```

**Comment by multiplegeorges** [comment]
Can you give it a try with the current HEAD of master?

**Comment by jkwasniak** [comment]
```sh
mac$ yarn deploy-development
yarn run v1.12.1
$ AWS_PROFILE=development vue-cli-service s3-deploy --mode development
{"assetPath":"dist","bucket":"www.xxxxxxx.de","region":"eu-central-1","deployPath":"/","pwa":false,"enableCloudfront":false,"uploadConcurrency":5,"awsProfile":"default","createBucket":false,"staticHosting":true,"staticIndexPage":"index.html","staticErrorPage":"index.html","assetMatch":"**","acl":"public-read","pluginVersion":"3.0.0"}
 INFO  Options: {"assetPath":"dist","bucket":"www.xxxxxxx.de","region":"eu-central-1","deployPath":"/","pwa":false,"enableCloudfront":false,"uploadConcurrency":5,"awsProfile":"development","createBucket":false,"staticHosting":true,"staticIndexPage":"index.html","staticErrorPage":"index.html","assetMatch":"**","acl":"public-read","pluginVersion":"3.0.0"}
 ERROR  Error: Service must be constructed with `new' operator
Error: Service must be constructed with `new' operator
    at Object.Service (/node_modules/aws-sdk/lib/service.js:25:28)
    at Object.features.constructor [as S3] (/node_modules/aws-sdk/lib/util.js:602:24)
    at module.exports (/node_modules/vue-cli-plugin-s3-deploy/s3deploy.js:176:12)
    at internalTickCallback (internal/process/next_tick.js:77:7)
    at process._tickCallback (internal/process/next_tick.js:47:5)
    at Function.Module.runMain (internal/modules/cjs/loader.js:763:11)
    at startup (internal/bootstrap/node.js:308:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:878:3)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by multiplegeorges** [comment]
Oh yeah. Fixed.

**Comment by jkwasniak** [comment]
Thanks! Now both variants appear to work
```sh
"deploy-1": "AWS_PROFILE=development vue-cli-service s3-deploy --mode development",
"deploy-2": "vue-cli-service s3-deploy --mode development",
```
Will there be Version 3.0.1 with the fix?

**Comment by jkwasniak** [comment]
Currently, there seem to be some problems.

First call yarn deploy - development will be executed successfully, but it will not transfer files.

```
yarn deploy-development
yarn run v1.12.3
{"assetPath":"dist","bucket":"www.xxxxxxxx.de","region":"eu-central-1","deployPath":"/","pwa":false,"enableCloudfront":false,"uploadConcurrency":5,"awsProfile":"default","createBucket":false,"staticHosting":true,"staticIndexPage":"index.html","staticErrorPage":"index.html","assetMatch":"**","acl":"public-read","pluginVersion":"3.0.0"}
AWS credentials confirmed
 INFO  Bucket: www.xxxxxxxx.de exists.
 INFO  Static Hosting is enabled.
✔ Deployment directory created.
✔ All 0 assets have been successfully deployed to https://s3-eu-central-1.amazonaws.com/www.xxxxxxxx.de/
```

The second call yarn deploy development will be successfully executed and it will transfer all files.

```
yarn deploy-development
yarn run v1.12.3
{"assetPath":"dist","bucket":"www.xxxxxxxx.de","region":"eu-central-1","deployPath":"/","pwa":false,"enableCloudfront":false,"uploadConcurrency":5,"awsProfile":"default","createBucket":false,"staticHosting":true,"staticIndexPage":"index.html","staticErrorPage":"index.html","assetMatch":"**","acl":"public-read","pluginVersion":"3.0.0"}
AWS credentials confirmed
 INFO  Bucket: www.xxxxxxxx.de exists.
 INFO  Static Hosting is enabled.
✔ Deployment directory created.
...
✔ ✔ All 53 assets have been successfully deployed to https://s3-eu-central-1.amazonaws.com/www.xxxxxxxx.de/
 INFO  Deployment complete.
```
**Other question:** The plugin now produces during the execution a directory with name **dist-deploy** (the same content as the **dist** directory). What's the **dist-deploy** for?

**Comment by robert-moore** [comment]
Just ran into the same issue. I selected an AWS Profile in the setup, but the ENV AWS credentials overrode this option

```
s3Deploy: {
      awsProfile: 'floom',
     ...
}
```

**Comment by vincent-lu** [comment]
Having similar problem; s3-deploy is ignoring the VUE_APP_S3D_AWS_PROFILE setting in .env.mode files when running `yarn deploy --mode development`.

Running with `AWS_PROFILE=development vue-cli-service s3-deploy --mode development` is interesting, first when checking if the bucket it seems to be checking against the right profile, but when trying to create bucket / upload I got permission denied error (most likely due to s3-deploy is trying to upload via the default aws profile instead development profile).

I'm running on s3-deploy 3.0.0.

**Comment by multiplegeorges** [comment]
@vincent-lu  I have an in-progress branch at https://github.com/multiplegeorges/vue-cli-plugin-s3-deploy/tree/refactor-plus-tests 

You could try checking that out and linking it in.
I'm going to release it soon, but your testing would be helpful.

**Comment by jkwasniak** [comment]
I have try to integrated refactor-plus-tests branch
"devDependencies": {
  "vue-cli-plugin-s3-deploy": "git://github.com/multiplegeorges/vue-cli-plugin-s3-deploy.git/#refactor-plus-tests",
}

After yarn install and vue-cli-service build --mode development 

yarn run v1.13.0
$ vue-cli-service build --mode development
internal/modules/cjs/loader.js:611
    throw err;
    ^

Error: Cannot find module 'vue-cli-plugin-s3-deploy'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:609:15)
    at Function.Module._load (internal/modules/cjs/loader.js:535:25)
    at Module.require (internal/modules/cjs/loader.js:663:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at idToPlugin (/Users/janusz/Documents/Developer/Javascript/xxx/xxx-web/node_modules/@vue/cli-service/lib/Service.js:136:14)
    at Object.keys.concat.filter.map.id (/Users/janusz/Documents/Developer/Javascript/xxx/xxx-web/node_modules/@vue/cli-service/lib/Service.js:176:20)
    at Array.map (<anonymous>)
    at Service.resolvePlugins (/Users/janusz/Documents/Developer/Javascript/xxx/xxx-web/node_modules/@vue/cli-service/lib/Service.js:162:10)
    at new Service (/Users/janusz/Documents/Developer/Javascript/filecq/xxx-web/node_modules/@vue/cli-service/lib/Service.js:33:25)
    at Object.<anonymous> (/Users/janusz/Documents/Developer/Javascript/xxx/xxx-web/node_modules/@vue/cli-service/bin/vue-cli-service.js:16:17)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

Did I do something wrong?

**Comment by multiplegeorges** [comment]
@jkwasniak I added the built files so it'll work now. 

Also, you can shorten your line to:

```json
"vue-cli-plugin-s3-deploy": "multiplegeorges/vue-cli-plugin-s3-deploy#refactor-plus-tests"
```

**Comment by jkwasniak** [comment]
Okay, thanks, installation is going
Now: Error: Cannot find module '@babel/polyfill'

```
MacBook-Pro:xxx-web janusz$ yarn build-development
yarn run v1.13.0
$ vue-cli-service build --mode development
internal/modules/cjs/loader.js:611
    throw err;
    ^

Error: Cannot find module '@babel/polyfill'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:609:15)
    at Function.Module._load (internal/modules/cjs/loader.js:535:25)
    at Module.require (internal/modules/cjs/loader.js:663:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/Users/janusz/Documents/Developer/Javascript/xxx/xxx-web/node_modules/vue-cli-plugin-s3-deploy/index.js:3:1)
    at Module._compile (internal/modules/cjs/loader.js:734:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:745:10)
    at Module.load (internal/modules/cjs/loader.js:626:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:566:12)
    at Function.Module._load (internal/modules/cjs/loader.js:558:3)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by jkwasniak** [comment]
after add @babel/polyfill to devDependencies (It's temporary solution)
```
"devDependencies": {
    "@babel/polyfill": "^7.2.5",
    "vue-cli-plugin-s3-deploy": "multiplegeorges/vue-cli-plugin-s3-deploy#refactor-plus-tests"
  },
```
and 
```
vue invoke s3-deploy
```
it seems as if the plugin works

**Comment by robindierckx** [comment]
@jkwasniak     "vue-cli-plugin-s3-deploy": "multiplegeorges/vue-cli-plugin-s3-deploy#refactor-plus-tests"
isn't available anymore. Has it been merged somewhere ?

**Comment by nicekiwi** [comment]
@robindierckx #refactor-plus-tests was merged into master and is now 4.0.0-rc3.

**Comment by nicekiwi** [comment]
Documentation for correctly setting the production deployment modes has been added in the latest devel branch.
